docs/design/graceful-domain-failover/3051-graceful-domain-failover.md

```diff
@@ -0,0 +1,239 @@
+# Design doc: Cadence graceful domain failover
+
+Author: Cadence Team
+
+Last updated: Mar 2020
+
+Reference: [#3051](https://github.com/uber/cadence/issues/3051)
+
+
+## Abstract
+
+Cadence supports domain failover with multi-cluster setup. However, the problems with the current failover are:
+1. Workflow progress could be lost.
+2. No causal consistency guarantee on domain level.
+
+The graceful domain failover uses to solve those two problems.
+
+## Use cases
+
+Users can trigger graceful domain failover via Cadence CLI (timeout is optional):
```

Comment:
I am confused why timeout is optional - it seems like the below design depends on a timeout? Do we provide a default if one is not given?

Reply:
Yes. We provides a default max support failover timeout